### PR TITLE
Adding BP_UBI_RUN_IMAGE_OVERRIDE environment variable

### PR DESCRIPTION
**Pull request by pacostas**
## Merge After
@paketo-bot-reviewer 
* https://github.com/paketo-community/ubi-nodejs-extension/pull/82

## Summary

This PR introduces an new environment variable called `BP_UBI_RUN_IMAGE_OVERRIDE`. The purpose of this env variable, is to set the run image with the value of `BP_UBI_RUN_IMAGE_OVERRIDE`. 

On the generate phase of the extension, is specified which image should be set as a run image. With this PR, we are able to explicitly set the run image according to the value of `BP_UBI_RUN_IMAGE_OVERRIDE`. As a result if the `BP_UBI_RUN_IMAGE_OVERRIDE` environment variable has been set and is not an empty string, the run image will have the value of `BP_UBI_RUN_IMAGE_OVERRIDE` otherwise is determined on how it used to be in the past, based on the node version.

## Use Cases

This change, helps us on setting run images without modifying the source code of the extension (the extension.toml file). E.x. when we release a new stack, we dont have the option to test the release candidate stack without modifying the source code of the extension, as the run image is not published yet to a registry. By explicitly specifying the run image with the `BP_UBI_RUN_IMAGE_OVERRIDE`, we are able to test that run image, before creating a release and pushing it to a registry.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: generate.go
Comment (mhdawson) [inline generate.go:59]: There should be a space after the // in comments, ame goes for other new comments as well

**Review by mhdawson — APPROVED** [review]
Overall LGTM, just need to add space after // in comment lines.   Believe that is what is used in existing comments (at least from my look at the nodejs-engine)

File: generate.go
Comment (pacostas) [reply generate.go]: I force pushed with the changes to keep the same history

**Comment by pacostas** [comment]
@paketo-bot-reviewer Tests are failing as the https://github.com/paketo-community/ubi-nodejs-extension/pull/82 hasnt been merged yet.

**Review by thitch97 — APPROVED** [review]

**Comment by thitch97** [comment]
@pacostas this all makes sense to me. Happy to merge it in once the merge conflicts are resolved

**Pull request merged**
